internal/graph/graph_test.go

```diff
@@ -708,6 +708,74 @@ func TestConnectedObjectGraph_RelationshipIngresss(t *testing.T) {
 				},
 			},
 		},
+		{
+			name: "IngressesInvolvingWildcardInTypes",
+			model: &openfgapb.AuthorizationModel{
+				TypeDefinitions: []*openfgapb.TypeDefinition{
+					{
+						Type: "user",
+					},
+					{
+						Type: "document",
+						Relations: map[string]*openfgapb.Userset{
+							"editor": typesystem.This(),
+							"viewer": typesystem.ComputedUserset("editor"),
+						},
+						Metadata: &openfgapb.Metadata{
+							Relations: map[string]*openfgapb.RelationMetadata{
+								"editor": {
+									DirectlyRelatedUserTypes: []*openfgapb.RelationReference{
+										typesystem.WildcardRelationReference("user"),
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			target: typesystem.DirectRelationReference("document", "viewer"),
+			source: typesystem.DirectRelationReference("user", ""),
+			expected: []*RelationshipIngress{
+				{
+					Type:    DirectIngress,
+					Ingress: typesystem.DirectRelationReference("document", "editor"),
+				},
+			},
+		},
+		{
+			name: "IngressesInvolvingWildcardInSource",
```

Comment:
This behavior needs to change just slightly.. `user` is directly related with `user:*` but `user:*` is not directly related with `user`. Typed wildcard connectivity is not bi-directional.

For example, consider the following model:
```
type user

type document
  relations
    define viewer: [user]
```
If you call `ListObjects(document#viewer, user:*)` then all relationship tuples related directly to a user can be pruned out, and so in terms of the graph of connectivity they are not connected at all. Put different, you cannot expand a subgraph relating `document#viewer` relationships to a `user` through `user:*`.

However, the opposite is not true. If you have the model

```
type user

type document
  relations
    define viewer: [user:*]
```
If you call `ListObjects(document#viewer, user:jon)` then all relationship tuples related directly to a `user:*` _cannot_ be pruned out. In terms of the graph connectivity they are directly connected. You can (and must) expand all subgraphs relating `document#viewer` to `user:*` in order to determine connectivity with all user relationships.